cql3/expr/restrictions.cc

```diff
@@ -241,6 +241,29 @@ binary_operator validate_and_prepare_new_restriction(const binary_operator& rest
             format("expr::validate_and_prepare_new_restriction unhandled restriction: {}", prepared_binop));
     }
 
+    // Convert single element IN relation to an EQ relation
```

Comment:
I'm not familiar with this code, so I have to ask: Why do we bother with this optimization (and risk of introducing bugs in it - do we have tests for single-item IN?)? I understand that a single-element IN is perhaps slightly less efficient than =, but is the difference substantial?